proxy.go

```diff
@@ -360,14 +360,18 @@ func (p *Proxy) RefreshDestinations(serviceName string, ring *consistent.Consist
 		return
 	}
 
-	// At the last moment, lock the mutex and defer so we unlock after setting.
-	// We do this after we've fetched info so we don't hold the lock during long
-	// queries, timeouts or errors. The flusher can lock the mutex and prevent us
-	// from updating at the same time.
+	updateRing(destinations, ring, mtx)
+	samples.Add(ssf.Gauge("discoverer.destination_number", float32(len(destinations)), srvTags))
+}
+
+// updateRing lock the mutex and defer so we unlock after setting.
+// We do this after we've fetched info so we don't hold the lock during long
+// queries, timeouts or errors. The flusher can lock the mutex and prevent us
+// from updating at the same time.
+func updateRing(destinations []string, ring *consistent.Consistent, mtx *sync.Mutex) {
 	mtx.Lock()
 	defer mtx.Unlock()
```

Comment:
i realize this was here before, but if we care about performance enough to be talking about "At the last moment" in this segment, then better not to use a defer.

Reply:
Fixed.